src/Core/Search/Builder/ClassFiltersBuilder.php

```diff
@@ -62,6 +96,23 @@ public function buildFilters(Filters $filters = null)
             return $filters;
         }
 
+        $typedBuilder = $this->getTypedBuilder();
+        if (null !== $typedBuilder) {
+            $typedFilters = $typedBuilder->buildFilters($filters);
+        } else {
+            $typedFilters = $this->buildTypedFilters($filters);
+        }
```

Comment:
```suggestion
        $typedBuilder = $this->getTypedBuilder();
        $typedFilters = $this->buildTypedFilters($filters);
        
        if (null !== $typedBuilder) {
            $typedFilters = $typedBuilder->buildFilters($filters);
        } 
```

Reply:
btw @jolelievre why did you call it "ClassFiltersBuilder"? What does "class" stands here for?

Reply:
@PululuK I can't do that, the purpose is to avoid using the "regular" way of building when a specific builder is defined for this type. There are two reasons to avoid this:
- performance, although it's not a big drawback but if we don't plan on using it there's no reason to build the regular way
- some Filter class may have addition conditions or arguments required in constructor (this is the case here for `ProductCombinations` which requires a `product_id` filter, if not present it will throw an error So if we systematically build the filters without its dedicated builder the error will be thrown here

@zuk3975 this is old code, but the idea was that this builder uses the `filters_class` config to build the filters with the appropriate class (`MetaFilters` `AddressFilters` and so on instead of the default `Filters` class) That's why I used `Class` in its name, maybe it should have been named `TypedFiltersBuilder` instead?

Reply:
I just added some comment and tests to ensure this behaviour

Reply:
Sorry @jolelievre  I don't understand 😀
I just suggest 

```php
A = x;
if(null !== B) {
    A = y;
}
```

instead of 

```php
if(null !== B) {
    A = y;
} else  {
    A = x;
}
```

Why we can't do that ?

Reply:
done this: (class name changed, check TypedFiltersBuilder.php)
`return $typedBuilder ? $typedBuilder->buildFilters($filters) : $this->defaultBuilder->buildFilters($filters);`

Reply:
Hi @PululuK 

In your example if it's just assigning value it's ok, but it's a simple case. In this one your `A = x;` is `$typedBuilder = $this->getTypedBuilder();`

And we DON'T want to call `getTypedBuilder` because it's a function that create the object in a generic way, but in the case of specific Filters class it can't. Especially here `ProductCombinationFilters` will throw an exception if `product_id` is not provided, `product_id` is only accessible in the route attributes so it needs a specific builder which is `ProductCombinationFiltersBuilder` which retrieves the `product_id` from the request attributes, and then provide it correctly in the `ProductCombinationFilters` constructor.

To sum it up, when a specific Typed builder is available we use this one ONLY because it may be the only one able to build the specific Filters class correctly. When we don't have a specific builder we use the default builder instead. But we can't assume that the generic builder will work in all situations, that's why it can't be called systematically.

Reply:
Yep, U'are right thanks guys ! LGTM

Reply:
I hope my explanation makes sense 😅  Don't hesitate to tell me if it's not!